### PR TITLE
feat: remove `link` without package name; add `link-assets` commands

### LINKED ISSUE #48: Make package name required for "link"
When you don't pass a package name to the "react-native link", it will try to look for all packages to link from your node_modules.

This is causing us a lot of issues, such as linking wrong packages or the ones that someone doesn't want to link (e.g. decided to use a different way of linking that is out of scope for this command).

We should simply stick to requiring package name.

### PR DESCRIPTION
**Pull request by Esemesek**
Removes deprecated behavior of `react-native link` without providing `packageName`.
Fixes #48 

Update:
Added `link-assets` command to support linking project assets.

### CODE REVIEW

**Comment by thymikee** [comment]
@Esemesek mind rebasing?

**Comment by grabbou** [comment]
Let's rebase and get that merged. I would like to note that there's few people that use `link` to link assets and it would be great to provide them a workaround when we ship this.

**Comment by thymikee** [comment]
Oh, that's a valid point, we need to introduce `link-assets` (or so) before we can ship this

File: packages/cli/src/commands/link/__tests__/link-test.js [hunk 1]
```diff
@@ -20,9 +20,9 @@ describe('link', () => {
     jest.resetModules();
   });
 
-  it('should reject when run in a folder without package.json', done => {
+  it('should reject when run in a folder without package.json', () => {
```
Comment (thymikee) [inline packages/cli/src/commands/link/__tests__/link-test.js]:
```suggestion
  it('should throw when run in a folder without package.json', () => {
```

File: packages/cli/src/commands/link/link.js [hunk 2]
```diff
@@ -7,64 +7,52 @@
...
+    throw new ReactNativeNotFound();
   }
   const hasProjectConfig = Object.keys(platforms).reduce(
     (acc, key) => acc || key in project,
     false,
   );
   if (!hasProjectConfig && findReactNativeScripts()) {
     throw new Error(
-      '`react-native link [package]` can not be used in Create React Native App projects. ' +
+      '`react-native link <package>` can not be used in Create React Native App projects. ' +
```
Comment (thymikee) [inline packages/cli/src/commands/link/link.js]: CRNA is deprecated, let's remove this check

File: packages/cli/src/commands/link/linkAssets.js [hunk 3]
```diff
@@ -32,4 +38,33 @@ const linkAssets = (
...
+  ctx: ContextT,
+  opts: LinkFlagsType,
+) {
+  try {
+    const {platforms, project} = getPlatformsAndProject(ctx, opts);
+    const projectAssets = getAssets(ctx.root);
+
+    return linkAssets(platforms, project, projectAssets);
+  } catch (err) {
+    throw new ReactNativeNotFound();
```
Comment (thymikee) [inline packages/cli/src/commands/link/linkAssets.js]: this is quite brutal and eats all errors. The `ReactNativeNotFound` check should live inside `linkAssets`, same as in `link`.

File: packages/cli/src/commands/link/types.flow.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+// @flow
+
+export type LinkFlagsType = {
```
Comment (thymikee) [inline packages/cli/src/commands/link/types.flow.js]: Can we rename it to just `LinkOptions`?

File: packages/cli/src/commands/link/linkAssets.js (hunk 3 above)
Comment (Esemesek) [reply packages/cli/src/commands/link/linkAssets.js]: True. Didn't notice that the `Promise.reject` call had `err` carried over.

**Comment by Esemesek** [comment]
Closing in favor of auto-linking.

**Pull request closed without merging**